test/unicode/utf8.jl

```diff
@@ -1,43 +1,48 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 ## Test for CESU-8 sequences
-
-let ch = 0x10000
-    for hi = 0xd800:0xdbff
-        for lo = 0xdc00:0xdfff
-            @test convert(String, Vector{UInt8}(String(Char[hi, lo]))) == string(Char(ch))
-            ch += 1
+@testset "cesu8 input" begin
+    let ch = 0x10000
+        for hi = 0xd800:0xdbff
+            for lo = 0xdc00:0xdfff
+                @test convert(String, Vector{UInt8}(String(Char[hi, lo]))) == string(Char(ch))
+                ch += 1
+            end
         end
     end
 end
 
-
-let str = String(b"this is a test\xed\x80")
-    @test next(str, 15) == ('\ufffd', 16)
-    @test_throws BoundsError getindex(str, 0:3)
-    @test_throws BoundsError getindex(str, 17:18)
-    @test_throws BoundsError getindex(str, 2:17)
-    @test_throws UnicodeError getindex(str, 16:17)
-    @test string(Char(0x110000)) == "\ufffd"
-    @test convert(String, b"this is a test\xed\x80\x80") == "this is a test\ud000"
+@testset "string indexing" begin
+    let str = String(b"this is a test\xed\x80")
```

Comment:
The `let` actually isn't necessary now that it's in a `testset`, since `str` is scoped within the `testset`. Not that you need to change it, just letting you know. Doesn't hurt to keep the `let`.